### PR TITLE
Hotfix / Improve feed performance

### LINKED ISSUE #1: Improve scores feed performance
When loading the scores feed we are running into the [n+1 queries problem](https://blog.appsignal.com/2018/04/24/active-record-performance-the-n-1-queries-antipattern.html)

Screenshot_2021-07-05_at_20 34 52

The fix must be done in a hotfix branch.

Make sure that `rubocop` and `rspec` still pass.

### PR DESCRIPTION
**Pull request by andreidumitru11**
Fix: [#1](https://github.com/AACraiu/golfr_backend/issues/1) Improve scores feed performance

**Changes**

We load the users in two queries  instead of n + 1. ( we would write a query that selects all the users that are in the vector of ids obtained from scores, but the  "includes" method does that under the hood )

**Before**
Screenshot 2024-07-29 at 13 55 09

**After**
Screenshot 2024-07-29 at 13 55 38

### CODE REVIEW

**Comment by cristian-scarisoreanu** [comment]
- The title should usually be **Hotfix / [Name of the issue]**

- You should add the issue URL to the PR

- You should follow the PR template:

image

- Maybe you could try to screenshot only the terminal, not the whole screen

File: Gemfile.lock [hunk 1]
```diff
@@ -1,190 +1,209 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    actioncable (6.1.6.1)
-      actionpack (= 6.1.6.1)
-      activesupport (= 6.1.6.1)
+    actioncable (6.1.7.8)
```
Comment (cristian-scarisoreanu) [inline Gemfile.lock]: There is no need to modify `Gemfile.lock`. You can remove it from the PR. Next time, please ensure this file is not in the committed files.

**Comment by cristian-scarisoreanu** [comment]
Fix rubocop warnings.

**Review by cristian-scarisoreanu — CHANGES_REQUESTED** [review]

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -69,12 +69,6 @@
   # Many RSpec users commonly either run the entire suite or an individual
   # file, and it's useful to allow more verbose output when running an
   # individual spec file.
-  if config.files_to_run.one?
```
Comment (cristian-scarisoreanu) [inline spec/spec_helper.rb]: Why did you remove this?

**Review by cristian-scarisoreanu — CHANGES_REQUESTED** [review]

File: spec/spec_helper.rb (hunk 2 above)
Comment (andreidumitru11) [reply spec/spec_helper.rb]: When I ran "rubocop" , it pointed out that the if statement it's empty.

Comment (cristian-scarisoreanu) [reply spec/spec_helper.rb]: Let's only modify the necessary files to solve the issue and revert this file to its original state. Ensure that rubocop does not add warnings to our modifications, not the entire codebase.

**Review by cristian-scarisoreanu — APPROVED** [review]

**Pull request merged**